### PR TITLE
Configurable key sizes for impersonating x509

### PR DESCRIPTION
**Pull request by nestorsokil**
## Fixes https://app.clubhouse.io/vgs/story/44078/vgs-create-stronger-server-certificate-for-sandbox-and-live-vault-environments

### CODE REVIEW

**Review by k-sever — APPROVED** [review]

**Comment by nestorsokil** [comment]
[ch60504]

**Comment by nestorsokil** [comment]
[ch44078]

File: src/main/java/org/littleshoot/proxy/mitm/CertificateSettings.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    private int defaultRootKeySize = DEFAULT_ROOT_KEY_SIZE;
+    private int defaultFakeKeySize = DEFAULT_FAKE_KEY_SIZE;
+
+    public CertificateSettingsBuilder setDefaultRootKeySize(int defaultRootKeySize) {
+      checkKeySize(defaultRootKeySize);
+      this.defaultRootKeySize = defaultRootKeySize;
+      return this;
+    }
+
+    public CertificateSettingsBuilder setDefaultFakeKeySize(int defaultFakeKeySize) {
```
Comment (osklyarenko) [inline src/main/java/org/littleshoot/proxy/mitm/CertificateSettings.java:49]: let's not use default here - it's confusing

**Review by osklyarenko — CHANGES_REQUESTED** [review]
see my comments.

**Review by osklyarenko — APPROVED** [review]

**Pull request merged**